build.gradle

```diff
@@ -60,3 +61,42 @@ jacocoTestReport {
 }
 
 build.dependsOn jacocoTestReport
+
+pmd {
+    toolVersion = '5.4.2'
+    ignoreFailures = true
+    sourceSets = [sourceSets.main]
+    ruleSets = []
+    ruleSetFiles = files('pmd.xml')
+}
+
+task('pmd', type: Pmd, dependsOn: 'assemble') {
+    description = 'Run pmd'
+    group = 'verification'
+
+    reports {
+        html.enabled = true
+        xml.enabled = true
+    }
+}
+
+build.dependsOn 'pmd'
```

Comment:
I'd leave it this way in case the pmdPrint gets replaced with a better feedback mechanisms and the dependency chain won't get broken easily.